test_submission_schema/schemas/v1.json

```diff
@@ -0,0 +1,185 @@
+        "results"
+    ],
+    "properties": {
+        "version": {
+            "const": 1,
+            "description": "The version of the test submission schema."
+        },
+        "origin": {
+            "type": "object",
+            "required": [
+                "name",
+                "version",
+                "packaging"
+            ],
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "description": "The application used to run the test plan.",
+                    "examples": ["YARF", "Checkbox"]
+                },
+                "version": {
+                    "type": "string",
+                    "description": "The version of the application used to run the test plan.",
+                    "examples": ["1.0.0"]
+                },
+                "packaging": {
+                    "type": "object",
+                    "properties": {
+                        "type": {
+                            "type": "string",
+                            "description": "The type of application.",
+                            "enum": ["snap", "debian", "source"]
+                        },
+                        "name": {
+                            "type": "string",
+                            "description": "The name of the package installed on the system.",
+                            "examples": ["yarf", "checkbox"]
+                        },
+                        "version": {
+                            "type": "string",
+                            "description": "The version of the package installed on the system.",
+                            "examples": ["1.0.0"]
+                        },
+                        "revision": {
+                            "type": "string",
+                            "description": "The revision number of the installed Snap.",
+                            "examples": ["100", "150"]
+                        }
+                    },
+                    "if": {"properties": {"type": {"const": "snap"}}},
+                    "then": {"required": ["type", "name", "version", "revision"]},
+                    "else": {
+                        "if": {"properties": {"type": {"const": "debian"}}},
+                        "then": {"required": ["type", "name", "version"]},
+                        "else": {"required": ["type", "version"]}
+                    }
+                }
+            }
+        },
+        "session_data": {
+            "type": "object",
+            "required": [
+                "title",
+                "test_plan_id"
+            ],
+            "properties": {
+                "title": {
+                    "type": "string",
+                    "description": "The title of the test plan."
+                },
+                "test_plan_id": {
+                    "type": "string",
+                    "pattern": "^[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+::[a-zA-Z0-9_]+$",
```

Comment:
Test plans in Checkbox use more special characters like in `client-cert-desktop-n+1` and `server-full-24.04`. I didn't see any test plan using `/` but slash may also be a valid character. 
@pieqq is there a set of valid characters in Checkbox parser?

Reply:
From discussion in MM, let's make it free form for now.